hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -171,9 +172,16 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
             false /* throw NoSuchObjectException if the table doesn't exist */);
         return null;
       });
-
       if (purge && lastMetadata != null) {
         CatalogUtil.dropTableData(ops.io(), lastMetadata);
+
+        boolean tableOwnsLocation = PropertyUtil.propertyAsBoolean(catalogProperties,
+            CatalogProperties.DROP_FULL_TABLE_FOLDER, CatalogProperties.DROP_FULL_TABLE_FOLDER_DEFAULT);
+        if (tableOwnsLocation) {
+          fileIO.deleteFolder(lastMetadata.location());
```

Comment:
Maybe we just remove the empty `data` and `metadata` folders, and then remove the table folder?

Reply:
That would work, I also gave it a try. But we can't assume that only this table puts files into these folders, so if there is something else non-related to this table inside these tables than the delete operation would fail. I found it more errorproof to recursively drop the root folder of the table.

Reply:
That's true. There is one thing I wasn't sure about. I saw some old issues (also related to dropTable) where the root table dir not only contained a data/ and metadata/ folder but also the different partitions had some own directories under data/.
I'm not sure this could still happen, and I didn't manage to repro such a case but if in fact there are tables with such a directory structure the proposed approach wouldn't be able to drop the root dir then.

Reply:
For the record: we should check partitioned tables as well. I expect that the partition directories will remain as well